### PR TITLE
HW2

### PR DESCRIPTION
**Pull request by eshcherbin**

### CODE REVIEW

File: laughter/laughter_classification/rnn_laugh_classifier.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        self.hidden_size2 = hidden_size2
+        self.lstm2 = nn.LSTM(input_size2, hidden_size2)
+        self.fc2 = nn.Linear(hidden_size1 + hidden_size2, 2)
+
+        self.init_hidden()
+
+    def init_hidden(self, cuda=False):
+        if cuda:
+            self.hidden1 = (
+                torch.zeros(1, 1, self.hidden_size1).cuda(),
```
Comment (Rauf-Kurbanov) [inline laughter/laughter_classification/rnn_laugh_classifier.py]: You can use `.to(device)` instead of this if statement

File: laughter/laughter_classification/train_rnn_laugh_classifier.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    from laughter_classification.sspnet_data_sampler import SSPNetDataSampler
+    s = SSPNetDataSampler(CORPUS_ROOT)
+    return s.create_sampled_df(FRAME_LEN_SEC, save_path=dataset_file,
+                               force_save=True)
+
+
+def prepare_dataset(dataset):
+    snames = dataset.SNAME.unique()
+    train_snames, test_snames = train_test_split(snames, random_state=SEED)
+    xs_train = [dataset[dataset.SNAME == sname].iloc[:, :-2].values for sname in train_snames]
```
Comment (Rauf-Kurbanov) [inline laughter/laughter_classification/train_rnn_laugh_classifier.py:34]: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.isin.html

**Review by Rauf-Kurbanov — CHANGES_REQUESTED** [review]
Good job, but I notices consistently poor AUC compared to accuracy
I suspect this is because of unbalanced training split, could you please try to fix that or comments on that?

File: laughter/laughter_classification/train_rnn_laugh_classifier.py (hunk 2 above)
Comment (eshcherbin) [reply laughter/laughter_classification/train_rnn_laugh_classifier.py:34]: I don't see how `DataFrame.isin()` woud help here. I don't need to just find all rows with `SNAME` from `train_snames`, I also want to split the dataframe into training objects by `SNAME`.

File: laughter/laughter_classification/rnn_laugh_classifier.py (hunk 1 above)
Comment (eshcherbin) [reply laughter/laughter_classification/rnn_laugh_classifier.py]: Fixed

**Comment by eshcherbin** [comment]
No idea why the AUC is bad. The share of positive labels in the test set is the same as in the train set. Moreover, splitting the dataset differently didn't help (I've tried the the same `train_test_split` parameters as @egor-bogomolov).

Also, I've tried increasing the hidden vector sizes, still no luck.

**Comment by Rauf-Kurbanov** [comment]
Well good enough, if you are interseted we can get into this later 3/4
